### PR TITLE
Make sure the subdirectory in the tmp folder exists

### PR DESCRIPTION
**Pull request by Toflar**
Seems like on some systems writing to the tmp dir is possible but if written to a sub directory, this requires to be present first.

### CODE REVIEW

**Review by bytehead — DISMISSED** [review]

File: core-bundle/src/Resources/contao/classes/Crawl.php [hunk 1]
```diff
@@ -86,8 +86,16 @@ public function run()
...
 		$jobId = Input::get('jobId');
 		$queue = $factory->createLazyQueue();
-		$debugLogPath = sys_get_temp_dir() . '/contao-crawl/' . $jobId . '_log.csv';
-		$resultCache = sys_get_temp_dir() . '/contao-crawl/' . $jobId . '.result-cache';
+		$crawLogsDir = sys_get_temp_dir() . '/contao-crawl';
+
+		// Make sure the subdirectory exists so logs can be written
+		if (!is_dir($crawLogsDir))
+		{
+			mkdir($crawLogsDir);
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/classes/Crawl.php]: We should use `Filesystem::mkdir()` here. More robust implementation and better error handling.

**Review by leofeyer — CHANGES_REQUESTED** [review]

**Review by m-vo — COMMENTED** [review]
Why not inject it as a dependency? This would allow mocking the call.

**Comment by Toflar** [comment]
Because that whole class is not a service. There's no way to register properly as maintenance module yet.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

**Pull request merged**